### PR TITLE
fix: Token header 

### PR DESCRIPTION
**Pull request by jokestax**
## Description
When making changes for argocd token,it gives 415 (media not supported error),added json header as we are passing json data of username and password

### CODE REVIEW

File: internal/controller/argocd.go [hunk 1]
```diff
@@ -129,9 +135,11 @@ func (clctrl *ClusterController) InitializeArgoCD() error {
 				8080,
 				argoCDStopChannel,
 			)
+
 			argoCDToken, err = argocd.GetArgoCDToken("admin", argocdPassword)
+
 			if err != nil {
-				return err
+				return fmt.Errorf("Error getting argoCDToken : %w", err)
```
Comment (patrickdappollonio) [inline internal/controller/argocd.go]: It's a bit of a norm in Go that error messages returned should not be capitalized the first letter. That's so the caller can choose to print that, so if you have something like your message, it'll render as:

```
Error: Error getting argoCDToken : foo bar baz
```

Without capitalization, it becomes more readable:

```
Error: error getting argoCDToken : foo bar baz
```

One of the linters we bring will have this check done automatically!

File: internal/controller/argocd.go [hunk 2]
```diff
@@ -46,7 +46,13 @@ func (clctrl *ClusterController) InstallArgoCD() error {
 			}
 		}
 
-		argoCDInstallPath := fmt.Sprintf("github.com:kubefirst/manifests/argocd/cloud?ref=%s", pkg.KubefirstManifestRepoRef)
+		var argoCDInstallPath string
+		if clctrl.CloudProvider == "digitalocean" {
+			argoCDInstallPath = "github.com:konstructio/manifests/argocd/cloud?ref=v1.1.0"
+		} else {
+			argoCDInstallPath = fmt.Sprintf("github.com:kubefirst/manifests/argocd/cloud?ref=%s", pkg.KubefirstManifestRepoRef)
+		}
```
Comment (patrickdappollonio) [inline internal/controller/argocd.go:54]: Not sure about this so feel free to ignore... How come _just_ DigitalOcean gets the new manifests from @konstructio and not everyone? (genuine curiosity lol)

Comment (jokestax) [reply internal/controller/argocd.go:54]: the digital ocean k8s upgrade needs argo v2.12 ,so the new version tag is on the charts that is on Konstruct org, the version upgrade may break for other clouds ;-; ;)

File: internal/controller/argocd.go (hunk 1 above)
Comment (jokestax) [reply internal/controller/argocd.go]: Done  :)

**Review by patrickdappollonio — APPROVED** [review]

**Pull request merged**